File: app/src/main/java/com/mc10inc/codetest/di/ApplicationComponent.java
package com.mc10inc.codetest.di;

import com.mc10inc.codetest.login.LoginActivity;
import com.mc10inc.codetest.studylist.StudyListActivity;
import com.mc10inc.codetest.studylist.StudyListViewModel;

import javax.inject.Singleton;

import dagger.Component;

/**
 * TODO
 *
 * @author Adam Stroud &#60;<a href="mailto:adam.stroud@gmail.com">adam.stroud@gmail.com</a>&#62;
 */
@Singleton
@Component(modules = {ApplicationModule.class})
public interface ApplicationComponent {
    public void inject(StudyListActivity target);
    public void inject(LoginActivity target);
    public void inject(StudyListViewModel target);
}

File: app/src/main/kotlin/com/mc10inc/codetest/di/ApplicationComponent.kt
package com.mc10inc.codetest.di

import com.mc10inc.codetest.login.LoginActivity
import com.mc10inc.codetest.studylist.StudyListActivity
import com.mc10inc.codetest.studylist.StudyListViewModel
import dagger.Component
import javax.inject.Singleton

/**
 * TODO
 *
 * @author Adam Stroud &#60;[adam.stroud@gmail.com](mailto:adam.stroud@gmail.com)&#62;
 */
@Singleton
@Component(modules = [(ApplicationModule::class)])
interface ApplicationComponent {
    fun inject(target: StudyListActivity)
    fun inject(target: LoginActivity)
    fun inject(taget: StudyListViewModel)
}